multicluster/controllers/multicluster/service_controller.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2021 Antrea Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package multicluster
+
+import (
+	"context"
+	"errors"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	mcsv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
+	"antrea.io/antrea/multicluster/controllers/multicluster/common"
+	"antrea.io/antrea/multicluster/controllers/multicluster/core"
+)
+
+// ServiceReconciler reconciles a Service object
+type ServiceReconciler struct {
+	client.Client
+	Scheme               *runtime.Scheme
+	remoteClusterManager *core.RemoteClusterManager
+}
+
+func NewServiceReconciler(
+	Client client.Client,
+	Scheme *runtime.Scheme,
+	remoteClusterManager *core.RemoteClusterManager) *ServiceReconciler {
+	reconciler := &ServiceReconciler{
+		Client:               Client,
+		Scheme:               Scheme,
+		remoteClusterManager: remoteClusterManager,
+	}
+	return reconciler
+}
+
+//+kubebuilder:rbac:groups="",resources=services,verbs=get;list;watch;delete
+
+// Service Reconcile will clean up MCS Service if no corresponding ResourceImport in leader cluster
+func (r *ServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```

Comment:
there is actually handle a special case when resourceimport is deleted in leader cluster but the controller in member cluster is crashed at that time and missed the delete event, then this can help to clean up those service when member controller is back

Reply:
Should not ResourceimportReconciler handle that when it restarts?

Reply:
ResourceimportReconciler is started by remote cluster manager, so it can only watch Object events from leader cluster, not local object event in member cluster.

Reply:
Question - then how we clean up ServiceImport in the member cluster, and we clean up ResourceImport in the leader cluster? We should not need a Controller for such clean up as it is just one time operation after startup?

Reply:
I am working on this part to add ServiceImport cleanup logic and yes, it one time operation, I will check if kubebuilder provide such mechanism to run it as one time process.